.github/workflows/test.yml

```diff
@@ -0,0 +1,129 @@
+---
+name: Run DTS tests
+on:
+  pull_request:
+    branches:
+      - 'main'
+jobs:
+  build-dts:
+    uses: ./.github/workflows/build.yml
+    with:
+      cacheless: false
+  run-tests:
+    name: Run DTS tests
+    if: ${{ github.head_ref == 'develop' && contains(join(needs.*.result, ','), 'success') }}
+    needs: build
+    runs-on:
+      labels: dts-builder
+    steps:
+      - name: Checkout OSFV repo
+        uses: actions/checkout@v4
+        with:
+          repository: 'Dasharo/open-source-firmware-validation'
+          path: 'open-source-firmware-validation'
+          submodules: 'recursive'
+          ref: 'develop'
+      - name: Run IPXE server
+        shell: bash
+        run: |
+          mkdir ipxe
+          cp build/tmp/deploy/images/genericx86-64/dts-base-image-genericx86-64.cpio.gz ipxe
+          cp build/tmp/deploy/images/genericx86-64/bzImage ipxe
+          echo -e "\n
+          #!ipxe\n
+          imgfetch --name file_kernel bzImage\n
+          imgfetch --name file_initrd dts-base-image-genericx86-64.cpio.gz\n
+          kernel file_kernel root=/dev/nfs initrd=file_initrd\n
+          boot" > ipxe/dts.ipxe
+          cd ipxe && python3-m http.server 4321 &
```

Comment:
Are we certain all of these background processes are always killed when workflow is interrupted, or we need to take care of this?

This is less of a problem on GitHub runners, as each time new VM is started, but on the self-hosted runner, we may end up with zombie processes if this is not the case.